crates/pathfinder/src/rpc/api.rs

```diff
@@ -1020,6 +1020,63 @@ impl RpcApi {
     //     todo!("Figure out where to take it from.")
     // }
 
+    /// Returns the contract's latest nonce.
+    ///
+    /// Not currently supported correctly as nonce's aren't implemented yet. In the mean time
+    /// returns "0x0" until starknet reaches 0.10 at which point it will return an error instead.
+    pub async fn get_nonce(&self, contract: ContractAddress) -> RpcResult<ContractNonce> {
+        // Check that contract actually exists..
+        let storage = self.storage.clone();
+        let span = tracing::Span::current();
+        let jh = tokio::task::spawn_blocking(move || {
+            let _g = span.enter();
+            let mut db = storage
+                .connection()
+                .context("Opening database connection")?;
+            let tx = db.transaction().context("Creating database transaction")?;
+
+            let exists = crate::storage::ContractsTable::exists(&tx, contract)
+                .context("Reading contract from database")?;
+
+            anyhow::Result::<_, anyhow::Error>::Ok(exists)
+        });
+        let exists = jh
+            .await
+            .context("Database read panic or shutting down")
+            .map_err(internal_server_error)?
+            .map_err(internal_server_error)?;
```

Comment:
```suggestion
            .context("Database read panic or shutting down")
            .and_then(|x| x)
            .map_err(internal_server_error)?;
```

This would be more in line, which I'd prefer, because it will be easier to refactor same structure, you wouldn't have to wonder why ... Aah but here the blocking closure  returns anyhow::Result instead of the usual RpcResult.

Even though it's quite crufty to keep returning RpcResult from within the blocking closure I think it's preferable *now* to do a single trick instead of multiple different tricks.